Source/Scene/Cesium3DTileset.js

```diff
@@ -0,0 +1,1654 @@
+     * @param {ShadowMode} [options.shadows=ShadowMode.ENABLED] Determines whether the tileset casts or receives shadows from each light source.
+     * @param {Boolean} [options.skipLODs=true] Determines if level-of-detail skipping optimization should be used.
+     * @param {Number} [options.baseScreenSpaceError=1024] The screen-space error that must be reached before skipping LODs
+     * @param {Number} [options.skipScreenSpaceErrorFactor=10] Multiplier defining the minimum screen space error to skip when loading tiles. Used in conjuction with skipLevels to determine which tiles to load.
+     * @param {Number} [options.skipLevels=1] Constant defining the minimum number of levels to skip when loading tiles. When it is 0, no levels are skipped. Used in conjuction with skipScreenSpaceErrorFactor to determine which tiles to load.
+     * @param {Boolean} [options.immediatelyLoadDesiredLOD=false] When true, do not progressively refine. Immediately load the desired LOD.
+     * @param {Boolean} [options.loadSiblings=false] Determines whether sibling tiles should be loaded when skipping levels-of-detail. When true, the siblings of any visible and downloaded tile are downloaded as well.
+     *
+     * @exception {DeveloperError} The tileset must be 3D Tiles version 0.0.  See https://github.com/AnalyticalGraphicsInc/3d-tiles#spec-status
+     *
+     * @example
+     * var tileset = scene.primitives.add(new Cesium.Cesium3DTileset({
+     *      url : 'http://localhost:8002/tilesets/Seattle'
+     * }));
+     *
+     * @see {@link https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/README.md|3D Tiles specification}
+     */
+    function Cesium3DTileset(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        var url = options.url;
+
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.string('options.url', url);
+        //>>includeEnd('debug');
+
+        var tilesetUrl;
+        var baseUrl;
+
+        if (getExtensionFromUri(url) === 'json') {
+            tilesetUrl = url;
+            baseUrl = getBaseUri(url, true);
+        } else if (isDataUri(url)) {
+            tilesetUrl = url;
+            baseUrl = '';
+        } else {
+            baseUrl = url;
+            tilesetUrl = joinUrls(baseUrl, 'tileset.json');
+        }
+
+        this._url = url;
+        this._baseUrl = baseUrl;
+        this._tilesetUrl = tilesetUrl;
+        this._root = undefined;
+        this._asset = undefined; // Metadata for the entire tileset
+        this._properties = undefined; // Metadata for per-model/point/etc properties
+        this._geometricError = undefined; // Geometric error when the tree is not rendered at all
+        this._gltfUpAxis = undefined;
+        this._processingQueue = [];
+        this._selectedTiles = [];
+        this._desiredTiles = new ManagedArray();
```

Comment:
I agree, I think that's part of the `private soup` we have going on in this PR.  There is super tight coupling of a lot of classes and post merge stuff should get refactored to not have so many "private" functions.